roles/zuul/files/jobs/bonnyci.yaml

```diff
@@ -0,0 +1,43 @@
+- publisher:
+    name: bonnyci-logs
+    publishers:
+      - scp:
+          site: bonnyci-scp
+          files:
+            - target: 'logs/$LOG_PATH'
+              source: '/opt/bonnyci/logs/**'
+              keep-hierarchy: true
+              copy-after-failure: true
+
+- job-template:
+    name: 'bonnyci-run-{pipeline}'
+    pipeline: '{pipeline}'
+    test_log_dir: /opt/bonnyci/logs
+    builders:
+      - zuul-git-prep
+      - shell: |
+            #!/bin/bash
+            if [[ ! -f .bonnyci/run.sh ]]; then
+                echo "ERROR: No test script found at .bonnyci/run.sh, skipping"
+                exit 1
+            elif [[ ! -x .bonnyci/run.sh ]]; then
+                echo "ERROR: Test script found at .bonnyci/run.sh but it is not executable"
+                exit 1
+            fi
+
+            export BONNYCI_TEST_PIPELINE={pipeline}
+            export BONNYCI_TEST_LOG_DIR={test_log_dir}
+
+            (cd .bonnyci/ && ./run.sh)
```

Comment:
Why change the working directory? I think it makes sense for the tests to run from the root of the repository, but that might be an argument to not put the entrypoint in a subdirectory.